src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -657,5 +663,69 @@ async def get_requirements_pex(request: RequirementsPexRequest, setup: PythonSet
     return pex_request
 
 
+@dataclass(frozen=True)
+class PexReqsRequest:
+    addresses: Addresses
+    interpreter_constraints: InterpreterConstraints | None = None
+
+
+
+@dataclass(frozen=True)
+class PexReqs:
+    requirements: Lockfile | PexRequirements
+    pexes: Iterable[Pex]
+
+
+@rule
+async def get_lockfile_subset(
+    request: PexReqsRequest,
+    python_setup: PythonSetup,
+) -> PexReqs:
+    if python_setup.enable_resolves:
+        chosen_resolve = await Get(
+            ChosenPythonResolve, ChosenPythonResolveRequest(request.addresses)
+        )
+        lock_path = chosen_resolve.lockfile_path
+        requirements_file_digest = await Get(
+            Digest,
+            PathGlobs(
+                [lock_path],
+                glob_match_error_behavior=GlobMatchErrorBehavior.error,
+                description_of_origin=chosen_resolve.description_of_origin,
+            ),
+        )
+        _digest_contents = await Get(DigestContents, Digest, requirements_file_digest)
+        lock_bytes = _digest_contents[0].content
```

Comment:
This is currently duplicated in the giant rule in `pex.py` currently :disappointed: Should we make a new rule?